app/src/test/java/org/simple/clinic/settings/SettingsEffectHandlerTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.simple.clinic.settings
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import io.reactivex.subjects.PublishSubject
+import org.junit.After
+import org.junit.Test
+import org.simple.clinic.patient.PatientMocker
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.None
+import org.simple.clinic.util.scheduler.TrampolineSchedulersProvider
+import org.simple.clinic.util.toOptional
+import java.util.UUID
+
+class SettingsEffectHandlerTest {
+
+  private val userSession = mock<UserSession>()
+  private val effectsSubject = PublishSubject.create<SettingsEffect>()
+
+  private val effectHandlerTest: TestObserver<SettingsEvent> = effectsSubject
+      .compose(SettingsEffectHandler.create(
+          userSession = userSession,
+          schedulersProvider = TrampolineSchedulersProvider()
+      ))
+      .test()
+
+  @After
+  fun tearDown() {
+    effectHandlerTest.dispose()
+  }
+
+  @Test
+  fun `user details must be fetched when the load user details event is received`() {
```

Comment:
Isn't this test for a single user, why are we testing the response for two different users?

Reply:
This test is meant to verify that if the user gets updated in the background, the updated details are published as a new event so the model can be updated. I'll rework the test to make that more clear.

Reply:
Done.